### PR TITLE
ceph: Remove old references to luminous and mimic versions

### LINKED ISSUE #8409: ssl setting for ceph dashboard is ignored in all supported ceph versions
* Bug Report

**Deviation from expected behavior:**
Documentation updates required.
> ssl: Whether to serve the dashboard via SSL, ignored on Ceph versions older than 13.2.2
from https://github.com/rook/rook/blob/master/Documentation/ceph-cluster-crd.md

Rook doesn't support v13 mimic anymore, so the ssl section of the dashboard configuration should be removed, as discussed here: https://github.com/rook/rook/pull/8401/files#r677843415

**Expected behavior:**
Ignored configuration shouldn't exist in rook documentation

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
References to luminous and mimic, and unnecessary references to nautilus were found in the docs that have been cleaned up and clarified. At the same time, the tests are updated to use more recent versions.

**Which issue is resolved by this Pull Request:**
Resolves #8409 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-common-issues.md [hunk 1]
```diff
@@ -824,7 +824,6 @@ They are cases where looking at Kubernetes logs is not enough for diverse reason
 
 So for each daemon, `dataDirHostPath` is used to store logs, if logging is activated.
 Rook will bindmount `dataDirHostPath` for every pod.
-As of Ceph Nautilus 14.2.1, it is possible to enable logging for a particular daemon on the fly.
```
Comment (leseb) [inline Documentation/ceph-common-issues.md:827]: Do we want to keep Nautilus until 1.8?

Comment (travisn) [reply Documentation/ceph-common-issues.md:827]: This was discussed in the community meeting and #7908 is the tracking issue. Originally the proposal was for 1.7, but this just gives a bit longer for people to upgrade and for downstream flexibility.

Comment (leseb) [reply Documentation/ceph-common-issues.md:827]: I know but maybe I wasn't clear. If Nautilus is expected to be removed in 1.8 and this PR will be backported (it has the label) then this line should stay.  Ami I missing something?

Comment (travisn) [reply Documentation/ceph-common-issues.md:827]: Oh the minimum version of nautilus is already 14.2.5, right? So this comment didn't seem necessary.

Comment (leseb) [reply Documentation/ceph-common-issues.md:827]: Ok right, since Nautilus is already the default it's fine removing it then.

**Review by leseb — APPROVED** [review]

**Pull request merged**